src/shared/components/common/state-radios.tsx

```diff
@@ -0,0 +1,62 @@
+import { I18NextService } from "@services/index";
+import { randomStr } from "@utils/helpers";
+import classNames from "classnames";
+import { NoOptionI18nKeys } from "i18next";
+import { linkEvent } from "inferno";
+import { Component } from "inferno";
+
+export interface State {
+  value: string;
+  i18n: NoOptionI18nKeys;
+}
+
+export interface StateRadioProps {
+  className?: string;
+  allStates: State[];
+  currentState: string;
+  onClickHandler(val: string): void;
+}
+
+export class StateRadio extends Component<StateRadioProps, object> {
+  constructor(props: any, context: any) {
+    super(props, context);
+  }
+
+  render() {
+    const radioId = randomStr();
+    return (
+      <div
+        className={classNames(
+          "btn-group btn-group-toggle flex-wrap",
```

Comment:
In general I'd like to start moving away from button groups, and use more dropdown selects. If you look at lemmy-ui on mobile, a lot of these button groups overflow in weird ways. Plus google's android app guidelines seems to recomment against them for the most part.